sources/updater/update_metadata/src/lib.rs

```diff
@@ -219,49 +233,77 @@ impl Manifest {
         Ok(())
     }
 
-    /// Adds a wave to update, returns number of matching updates for wave
-    pub fn add_wave(
+    /// Returns Updates matching variant, arch, and version
+    fn get_matching_updates(
         &mut self,
         variant: String,
         arch: String,
         image_version: Version,
-        bound: u32,
-        start: DateTime<Utc>,
-    ) -> Result<usize> {
-        let matching: Vec<&mut Update> = self
-            .updates
+    ) -> Vec<&mut Update> {
+        self.updates
             .iter_mut()
             // Find the update that exactly matches the specified update
             .filter(|update| {
                 update.arch == arch && update.variant == variant && update.version == image_version
             })
-            .collect();
-        let num_matching = matching.len();
-        for update in matching {
-            update.waves.insert(bound, start);
-        }
-        Self::validate_updates(&self.updates)?;
-        Ok(num_matching)
+            .collect()
     }
 
-    pub fn remove_wave(
+    /// Adds a vec of waves to update, returns number of matching updates for wave
+    // Wave format in `manifest.json` is slightly different from the wave structs
+    // provided to this function. For example, if two `UpdateWave` structs are
+    // passed to this function:
+    // [
+    //   UpdateWave { start_after: "1 hour", fleet_percentage: 1 },
+    //   UpdateWave { start_after: "1 day", fleet_percentage: 100},
+    // ]
+    //
+    // The resulting `waves` section of the applicable update looks like:
+    // waves: {
+    //   "0": "<UTC datetime of 1 hour from now>",
+    //   "20": "<UTC datetime of 1 day from now>"
+    // }
+    //
+    // This might look odd until you understand that the first wave begins
+    // at the time specified, and includes seeds 0-19, or 1%, of the seeds
+    // available (`MAX_SEED` in this file). The next wave begins at the time
+    // specified and includes seeds 20-MAX_SEED, or 100% of the rest of the
+    // seeds available.
```

Comment:
```suggestion
    // seeds available.  We do this so that the waves input can be more understandable for human operators, with times relative to when they start a release, but still have absolute times and seeds that are more understandable in our update code.
```
Helps to say *why* we do something that looks odd!